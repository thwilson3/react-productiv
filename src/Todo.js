import React from "react";

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority }
 *
 * { EditableTodo, TopTodo } -> Todo
 **/

function Todo({todo}) {

  const {title, priority, description} = todo;
  return (
      <div id={`todo-${todo.id}`} className="Todo">
        <div><b>{title}</b> <small>(priority: {priority})</small></div>
        <div><small>{description}</small></div>
      </div>
  );
}

export default Todo;
