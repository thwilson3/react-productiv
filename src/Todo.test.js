import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";
import { initialTodos } from "./App";


describe("todo", function () {
it("renders without crashing", function () {
  // this is a low-value test, but better than nothing
  render(<Todo todo={initialTodos[0]}/>);
});

it("renders proper elements", function() {
    const {title, description, priority} = initialTodos[0];
    const result = render(<Todo todo={initialTodos[0]}/>);

    expect(result.queryByText(title)).toBeInTheDocument();
    expect(result.queryByText(description)).toBeInTheDocument();
    expect(result.queryByText(`(priority: ${priority})`)).toBeInTheDocument();
})

it("matches snapshot", function () {
    const { container } = render(<Todo todo={initialTodos[0]}/>);
    expect(container).toMatchSnapshot();
  });
});