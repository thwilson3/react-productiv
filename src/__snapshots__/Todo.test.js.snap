// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Code!
      </b>
       
      <small>
        (priority: 
        2
        )
      </small>
    </div>
    <div>
      <small>
        Write some code
      </small>
    </div>
  </div>
</div>
`;

exports[`todo matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Code!
      </b>
       
      <small>
        (priority: 
        2
        )
      </small>
    </div>
    <div>
      <small>
        Write some code
      </small>
    </div>
  </div>
</div>
`;
